src/Core/Geographic/Coordinates.js

```diff
@@ -1,441 +1,280 @@
- * // returns : 24999549
- *
- * // or
- *
- * const position = { x: 20885167, y: 849862, z: 23385912 };
- * const coords = new Coordinates('EPSG:4978', position.x, position.y, position.z);  // Geocentric system
- * const coordinates = coords.as('EPSG:4326');  // Geographic system
- * coordinates.altitude(); // Altitude in geographic system
- * // returns : 24999548.046711832
- *
- * @return     {number} - The altitude of the position.
+ * new Coordinates('EPSG:4326', 2.33, 48.24, 24999549); //Geographic coordinates
  */
+class Coordinates {
+    /**
+     * @constructor
+     *
+     * @param {string} crs - A supported crs (see the `crs` property below).
+     * @param {number|Array<number>|Coordinates|THREE.Vector3} [v0=0] -
+     * x or longitude value, or a more complex one: it can be an array of three
+     * numbers, being x/lon, x/lat, z/alt, or it can be `THREE.Vector3`. It can
+     * also simply be a Coordinates.
+     * @param {number} [v1=0] - y or latitude value.
+     * @param {number} [v2=0] - z or altitude value.
+     */
+    constructor(crs, v0 = 0, v1 = 0, v2 = 0) {
+        this.isCoordinates = true;
+
+        CRS.isValid(crs);
+        this.crs = crs;
+
+        // Storing the coordinates as is, not in arrays, as it is
+        // slower (see https://jsbench.me/40jumfag6g/1)
+        this.x = 0;
+        this.y = 0;
+        this.z = 0;
+
+        // Normal
+        this._normal = new THREE.Vector3();
+
+        if (v0.length > 0) {
+            this.setFromArray(v0);
+        } else if (v0.isVector3 || v0.isCoordinates) {
+            this.setFromVector3(v0);
+        } else {
+            this.setFromValues(v0, v1, v2);
+        }
 
-Coordinates.prototype.altitude = function altitude() {
-    _assertIsGeographic(this.crs);
-    return this._values[2];
-};
+        this._normalNeedsUpdate = true;
+    }
 
-/**
- * Set the altiude.
- * @example coordinates.setAltitude(number)
- * @param      {number} - Set the altitude.
- */
+    /**
+     * Set the values of this Coordinates.
+     *
+     * @param {number} [v0=0] - x or longitude value.
+     * @param {number} [v1=0] - y or latitude value.
+     * @param {number} [v2=0] - z or altitude value.
+     *
+     * @return {Coordinates} This Coordinates.
+     */
+    setFromValues(v0 = 0, v1 = 0, v2 = 0) {
+        this.x = v0;
+        this.y = v1;
+        this.z = v2;
+
+        this._normalNeedsUpdate = true;
+        return this;
+    }
 
-Coordinates.prototype.setAltitude = function setAltitude(altitude) {
-    _assertIsGeographic(this.crs);
-    this._values[2] = altitude;
-};
+    /**
+     * Set the values of this Coordinates from an array.
+     *
+     * @param {Array<number>} array - An array of number to assign to the
+     * Coordinates.
+     * @param {number} [offset] - Optional offset into the array.
+     *
+     * @return {Coordinates} This Coordinates.
+     */
+    setFromArray(array, offset = 0) {
+        this.x = array[offset];
```

Comment:
No, because there is no point to have it separate across multiple method if you always need to call one specific method. At least that's what I think in term of performance, but I could be wrong.

Reply:
to check

Reply:
It is approximately the same in fact https://jsbench.me/0ajvz7nexn/1

Reply:
ok, keep the actual code.